app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java

```diff
@@ -194,10 +219,64 @@ public boolean onOptionsItemSelected(final MenuItem item) {
                     showSnackbar(mActivity, R.string.snackbar_no_pdfs_selected);
                 }
                 break;
+            case R.id.item_merge:
+                if (mViewFilesAdapter.getItemCount() > 1) {
+                    showSnackbar(mActivity, "merged");
+                    mergeFiles();
+                }
+                break;
         }
         return true;
     }
 
+    /**
+     * Merge the selected PDFs into a single PDF
+     * Show a dialog to allow user to enter file name
+     */
+    void mergeFiles() {
```

Comment:
This file already is quite big & has a lot of utilities. You can add it option in any existing file in utils package, or create a new one.